addons/event_crm/models/event_crm.py

```diff
@@ -0,0 +1,104 @@
+    registration_filter_domain = fields.Text(string="Registrations Domain", default='[]', required=True,
+        help="Filter the attendees that will or not generate leads.")
+
+    # Lead default_value fields
+    lead_type = fields.Selection([
+        ('lead', 'Lead'), ('opportunity', 'Opportunity')], string="Lead Type", required=True,
+        default=lambda self: 'lead' if self.env['res.users'].has_group('crm.group_use_lead') else 'opportunity',
+        help="Default lead type when this rule is applied.")
+    lead_sales_team_id = fields.Many2one('crm.team', string='Sales Team', help="Automatically assign the created leads to this Sales Team.")
+    lead_user_id = fields.Many2one('res.users', string='Salesperson', help="Automatically assign the created leads to this Salesperson.")
+    lead_tag_ids = fields.Many2many('crm.tag', string='Tags', help="Automatically add these tags to the created leads.")
+
+    def _create_leads(self, registration_ids):
+        """
+        Create leads based on lead creation type of the rule and apply to this leads
+        some prefilled values based on the rule and the registration.
+        """
+        for rule in self:
+            vals_list = []
+            registrations = registration_ids
+            if rule.registration_filter_domain != '[]':
+                registrations = registration_ids.search([('id', 'in', registration_ids.ids)] + safe_eval(rule.registration_filter_domain))
+            if registrations and rule._check_rule(registrations):
+                lead_values = {
+                    'user_id': rule.lead_user_id.id,
+                    'type': rule.lead_type,
+                    'team_id': rule.lead_sales_team_id.id,
+                    'tag_ids': rule.lead_tag_ids,
+                    'registration_rule_id': rule.id,
+                }
+                if rule.lead_creation_basis == 'attendee':
+                    for registration in registrations:
+                        values = lead_values.copy()
+                        values.update(registration._get_registration_lead_values(rule.lead_creation_basis))
+                        vals_list.append(values)
+                else:
+                    if len(registrations) > 1 or not registrations[0].main_registration_id:
+                        lead_values.update(registrations._get_registration_lead_values(rule.lead_creation_basis))
+                        vals_list.append(lead_values)
+                    else:
+                        # only one registration
+                        for lead in registrations.main_registration_id.lead_ids:
+                            if lead.registration_rule_id.lead_creation_basis == 'order':
+                                lead.description += registrations._get_lead_description()
+                                registrations.lead_ids += lead
+
+                if vals_list:
+                    self.sudo().env['crm.lead'].create(vals_list)
+
+    def _check_rule(self, registration_ids):
```

Comment:
A bit strange to take registrations (to rename by the way) and return a boolean. Why not filtered registrations based on rule ? Or it is called with always one registration ? Anyway always better to prepare batch version, so I would filter. Caller adapt its strategy according to the returned recordset.